funcx_endpoint/funcx_endpoint/endpoint/interchange.py

```diff
@@ -523,49 +347,33 @@ def _start_threads_and_main(self):
 
         self.initial_registration_complete = False
 
-        log.info(
-            "Attempting connection to client at {} on ports: {},{},{}".format(
-                self.client_address,
-                self.client_ports[0],
-                self.client_ports[1],
-                self.client_ports[2],
-            )
-        )
-
-        self._task_puller_thread = threading.Thread(
-            target=self.migrate_tasks_to_internal, args=(self._quiesce_event,)
-        )
-        self._task_puller_thread.start()
-
-        self._command_thread = threading.Thread(
-            target=self._command_server, args=(self._quiesce_event,)
+        self._task_puller_proc = self.migrate_tasks_to_internal(
+            self.connection_params,
+            self.endpoint_id,
+            self.pending_task_queue,
+            self._quiesce_event,
         )
-        self._command_thread.start()
 
         try:
             self._main_loop()
         except Exception:
             log.exception("[MAIN] Unhandled exception")
         finally:
             self.results_outgoing.close()
+            self._task_puller_proc.terminate()
```

Comment:
We went through the trouble of defining a clean `stop()` method in order to avoid using `terminate()`. Let's use it here.